src/Microsoft.Restier.Publishers.OData/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Web.OData;
+using System.Web.OData.Formatter.Deserialization;
+using Microsoft.OData.Core;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Library;
+using Microsoft.Restier.Core;
+
+namespace Microsoft.Restier.Publishers.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Get clr type from payload.
+    /// </summary>
+    internal static class DeserializationHelpers
+    {
+        internal static object ConvertValue(object oDataValue, Type expectedReturnType, ref IEdmTypeReference propertyType, ApiBase api)
+        {
+            var model = api.GetModelAsync().Result;
```

Comment:
I change these method to async and use await, this should avoid deadlock, let me know if you have other suggestions on this part. I am new to c# :)